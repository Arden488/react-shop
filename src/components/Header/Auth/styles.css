.header-auth svg {
  display: inline-block;
  vertical-align: middle;
}