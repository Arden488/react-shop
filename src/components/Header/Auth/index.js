import React, { Component } from 'react';
import feather from 'feather-icons';

import './styles.css';

class Auth extends Component {
  componentDidMount() {
    feather.replace();
  }

  render() {
    return (
      <div className="header-auth">
        <i data-feather="user" />
      </div>
    );
  }
}

export default Auth;
