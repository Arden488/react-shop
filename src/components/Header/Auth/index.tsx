import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import feather from 'feather-icons';
import Dropdown from '../../Dropdown';
import DropdownMenu from '../../Dropdown/DropdownMenu';

import './styles.css';

class Auth extends Component {
  componentDidMount() {
    feather.replace();
  }

  render() {
    return (
      <div className="header-auth">
        <Dropdown>
          <button>
            <span>User</span>
            <i data-feather="user" />
          </button>
          <DropdownMenu>
            <Link to="/profile">Profile</Link>
            <Link to="/signout">Sign Out</Link>
          </DropdownMenu>
        </Dropdown>
      </div>
    );
  }
}

export default Auth;
