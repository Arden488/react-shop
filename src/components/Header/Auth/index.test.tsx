/* global it, expect, shallow, beforeEach */

import React from 'react';

import Auth from './index';

let component = null;

beforeEach(() => {
  component = shallow(<Auth />);
});

it('renders correctly', () => {
  expect(component).toMatchSnapshot();
});
