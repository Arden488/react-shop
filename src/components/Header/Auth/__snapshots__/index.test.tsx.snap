// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="header-auth"
>
  <Dropdown>
    <button>
      <span>
        User
      </span>
      <i
        data-feather="user"
      />
    </button>
    <DropdownMenu>
      <Link
        replace={false}
        to="/profile"
      >
        Profile
      </Link>
      <Link
        replace={false}
        to="/signout"
      >
        Sign Out
      </Link>
    </DropdownMenu>
  </Dropdown>
</div>
`;
